solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseHttpClusterStateProvider.java

```diff
@@ -173,6 +167,33 @@ private ClusterState fetchClusterState(
     return cs;
   }
 
+  private SimpleOrderedMap<?> submitClusterStateRequest(
+      SolrClient client,
+      String collection,
+      boolean fetchLiveNodes,
+      boolean fetchClusterProp,
+      boolean fetchNodeRoles)
```

Comment:
Methods with tons of booleans is hard to read at the caller.  Would it be sufficiently concise to heave each caller request what they want?  I could imagine doing so in only a 2-3 lines each.

Reply:
Addressed in latest commit